vehicle_gateway_demo/src/vtol_position_control.cpp

```diff
@@ -0,0 +1,238 @@
+// Copyright 2023 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cmath>
+#include <iostream>
+
+#include <rclcpp/rclcpp.hpp>
+
+#include <pluginlib/class_loader.hpp>
+
+#include <vehicle_gateway_px4/vehicle_gateway_px4.hpp>
+
+#ifdef _WIN32
+#include <Windows.h>
+#else
+#include <unistd.h>
+#endif
+
+using namespace std;
+
+const double EARTH_RADIUS = 6378100;  // meters
+
+double calc_distance_latlon(double lat_1, double lon_1, double lat_2, double lon_2)
+{
+  // This uses a planar approximation, not the real trigonometry. This
+  // approximation is just fine for short distances, which is all we're
+  // currently considering.
+
+  double dx = (EARTH_RADIUS * M_PI / 180) * (lon_2 - lon_1) * cos(lat_1);
+  double dy = (EARTH_RADIUS * M_PI / 180) * (lat_2 - lat_1);
+
+  return sqrt(dx * dx + dy * dy);
+}
+
+class VehicleGatewayCpp
```

Comment:
This class has a lot of the same functionality as `VehicleGatewayPython`, as well as some new convenience methods. If we like this, it will be pretty easy to push a lot of this logic down into `vehicle_gateway_px4`.

Reply:
I would suggest to move `arm_sync` and `disarm_sync` to the `vehicle_gateway` interface. Then we will force all APIs to implement these methods which are very usefull. do you mind to add a TODO ?